src/controller_examples/rabbitmq_controller/exec/reconciler.rs

```diff
@@ -854,6 +853,29 @@ fn make_role_binding(rabbitmq: &RabbitmqCluster) -> (role_binding: RoleBinding)
     role_binding
 }
 
+fn update_stateful_set(rabbitmq: &RabbitmqCluster, mut found_stateful_set: StatefulSet) -> (stateful_set: StatefulSet)
+    requires
+        rabbitmq@.metadata.name.is_Some(),
+        rabbitmq@.metadata.namespace.is_Some(),
+    ensures
+        stateful_set@ == rabbitmq_spec::update_stateful_set(rabbitmq@, found_stateful_set@),
+{
+    let mut owner_references = Vec::new();
+    owner_references.push(rabbitmq.controller_owner_ref());
+    proof {
+        assert_seqs_equal!(
+            owner_references@.map_values(|owner_ref: OwnerReference| owner_ref@),
+            rabbitmq_spec::make_role(rabbitmq@).metadata.owner_references.get_Some_0()
+        );
+    }
+    let mut metadata = found_stateful_set.metadata();
+    metadata.set_owner_references(owner_references);
+    metadata.reset_finalizers();
```

Comment:
Can you add some comment on these set_owner_reference and reset_finalizers? Just to remind ourselves these are to make the proof go easier, without affecting the functionality.

Reply:
Done